### PR TITLE
 Add in errcheck, changelog-links from upstream terraform 

### PR DESCRIPTION
**Pull request by puneetk**
**Summary of Changes** 
* Remove BackTick
* Add in go errcheck script from terraform-provider-null
* Add in changelog script from terraform-provider-null

### CODE REVIEW

**Comment by webdevwilson** [comment]
Can you provide a better description of what these changes do? Especially the bit that sets git cookies. Thanks!

File: scripts/gofmtcheck.sh [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 # Check gofmt
 echo "==> Checking that code complies with gofmt requirements..."
-gofmt_files=$(gofmt -l `find . -name '*.go' | grep -v vendor`)
```
Comment (webdevwilson) [inline scripts/gofmtcheck.sh:5]: This is actually a script that is present in all the terraform provider repos. So I don't really feel the need to make changes to it.

File: scripts/changelog-links.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (webdevwilson) [inline scripts/changelog-links.sh:1]: We don't have a changelog at the moment. Nor are there any releases. I think it is fine if we want to do this and start tagging releases. But please check in the CHANGELOG.md as well.

**Comment by webdevwilson** [comment]
OK, I see what you did. Copied down updates from other providers. So other than the backtick to $() change I think the rest is good.

File: scripts/gofmtcheck.sh (hunk 1 above)
Comment (puneetk) [reply scripts/gofmtcheck.sh:5]: I agree, but Backticks must die.  Sent a PR here https://github.com/terraform-providers/terraform-provider-null/pull/11, we can wait on that one.

Comment (webdevwilson) [reply scripts/gofmtcheck.sh:5]: Why must backticks die?

Comment (puneetk) [reply scripts/gofmtcheck.sh:5]: Well `die` is a strong word, but especially in cases where we are nesting bash executions $() is preferred to backticks. Its more deterministic execution, and one does not have to worry about escaping and such. Ref http://mywiki.wooledge.org/BashFAQ/082

**Pull request merged**